### PR TITLE
fix(requirements): add support for non resource url rules

### PR DESCRIPTION
**Pull request by njhale**
### Description
- Adds support for NonResourceURL rules as CSV requirements
- Adds CRD served version CSV requirement check

Fixes: [ALM-755](https://jira.coreos.com/browse/ALM-755) and [ALM-756](https://jira.coreos.com/browse/ALM-756)

### CODE REVIEW

**Comment by eriknelson** [comment]
Tested with this patch against olm master and it's working great :+1:

File: pkg/controller/install/rule_checker.go [hunk 1]
```diff
@@ -155,3 +163,16 @@ func (c *CSVRuleChecker) hasOwnerConflicts(ownerRefs []metav1.OwnerReference) bo
 
 	return conflicts
 }
+
+func ruleValid(rule rbacv1.PolicyRule) error {
+	if len(rule.Verbs) == 0 {
```
Comment (ecordell) [inline pkg/controller/install/rule_checker.go]: does the particular openapi flavor kube supports let you specify `minItems` length on the verb list?

**Pull request merged**